Schema for `intConstantTileCodec` is incorrect.

The [schema](https://github.com/locationtech/geotrellis/blob/master/spark/src/main/scala/geotrellis/spark/io/avro/codecs/ConstantTileCodecs.scala#L183) says that the constant should be an array, but [the encoder](https://github.com/locationtech/geotrellis/blob/master/spark/src/main/scala/geotrellis/spark/io/avro/codecs/ConstantTileCodecs.scala#L183) writes a scalar. Causes unpleasant `Caused by: java.lang.ClassCastException: java.lang.Integer cannot be cast to java.util.Collection` error.

--- a/spark/src/main/scala/geotrellis/spark/io/avro/codecs/ConstantTileCodecs.scala
+++ b/spark/src/main/scala/geotrellis/spark/io/avro/codecs/ConstantTileCodecs.scala
@@ -180,7 +180,7 @@ trait ConstantTileCodecs {
       .fields()
       .name("cols").`type`().intType().noDefault()
       .name("rows").`type`().intType().noDefault()
-      .name("cell").`type`().array().items().intType().noDefault()
+      .name("cell").`type`().intType().noDefault()
       .name("noDataValue").`type`().unionOf().intType().and().nullType().endUnion().intDefault(Int.MinValue)
       .endRecord()
 

--- a/spark/src/test/scala/geotrellis/spark/io/avro/AvroTools.scala
+++ b/spark/src/test/scala/geotrellis/spark/io/avro/AvroTools.scala
@@ -65,6 +65,9 @@ object AvroTools {
   implicit class ShortNoDataValueCheckMethods(val self: ShortArrayTile) extends
     ShortNoDataChecker(self.cellType) with AvroNoDataCheckMethods[ShortArrayTile] {}
 
+  implicit class ShortConstantNoDataValueCheckMethods(val self: ShortConstantTile) extends
+    ShortNoDataChecker(self.cellType) with AvroNoDataCheckMethods[ShortConstantTile] {}
+
   class ShortNoDataChecker(cellType: CellType) extends NoDataValueChecker[ShortArrayTile] {
     override def doCheck(nodata: Option[Any]): Unit = {
       cellType match {
@@ -78,6 +81,9 @@ object AvroTools {
 
   implicit class UShortNoDataValueCheckMethods(val self: UShortArrayTile) extends
     UShortNoDataChecker(self.cellType) with AvroNoDataCheckMethods[UShortArrayTile] {}
+
+  implicit class UShortConstantNoDataValueCheckMethods(val self: UShortConstantTile) extends
+    UShortNoDataChecker(self.cellType) with AvroNoDataCheckMethods[UShortConstantTile] {}
 
   class UShortNoDataChecker(cellType: CellType) extends NoDataValueChecker[UShortArrayTile] {
     override def doCheck(nodata: Option[Any]): Unit = {
@@ -93,6 +99,9 @@ object AvroTools {
   implicit class IntNoDataValueCheckMethods(val self: IntArrayTile) extends
     IntNoDataChecker(self.cellType) with AvroNoDataCheckMethods[IntArrayTile] {}
 
+  implicit class IntConstantNoDataValueCheckMethods(val self: IntConstantTile) extends
+    IntNoDataChecker(self.cellType) with AvroNoDataCheckMethods[IntConstantTile] {}
+
   class IntNoDataChecker(cellType: CellType) extends NoDataValueChecker[IntArrayTile] {
     override def doCheck(nodata: Option[Any]): Unit = {
       cellType match {
@@ -106,6 +115,9 @@ object AvroTools {
 
   implicit class FloatNoDataValueCheckMethods(val self: FloatArrayTile) extends
     FloatNoDataChecker(self.cellType) with AvroNoDataCheckMethods[FloatArrayTile] {}
+
+  implicit class FloatConstantNoDataValueCheckMethods(val self: FloatConstantTile) extends
+    FloatNoDataChecker(self.cellType) with AvroNoDataCheckMethods[FloatConstantTile] {}
 
   class FloatNoDataChecker(cellType: CellType) extends NoDataValueChecker[FloatArrayTile] {
     override def doCheck(nodata: Option[Any]): Unit = {
@@ -124,6 +136,9 @@ object AvroTools {
   implicit class DoubleNoDataValueCheckMethods(val self: DoubleArrayTile) extends
     DoubleNoDataChecker(self.cellType) with AvroNoDataCheckMethods[DoubleArrayTile] {}
 
+  implicit class DoubleConstantNoDataValueCheckMethods(val self: DoubleConstantTile) extends
+    DoubleNoDataChecker(self.cellType) with AvroNoDataCheckMethods[DoubleConstantTile] {}
+
   class DoubleNoDataChecker(cellType: CellType) extends NoDataValueChecker[DoubleArrayTile] {
     override def doCheck(nodata: Option[Any]): Unit = {
       cellType match {
@@ -137,6 +152,9 @@ object AvroTools {
 
   implicit class ByteNoDataValueCheckMethods(val self: ByteArrayTile) extends
     ByteNoDataChecker(self.cellType) with AvroNoDataCheckMethods[ByteArrayTile] {}
+
+  implicit class ByteConstantNoDataValueCheckMethods(val self: ByteConstantTile) extends
+    ByteNoDataChecker(self.cellType) with AvroNoDataCheckMethods[ByteConstantTile] {}
 
   class ByteNoDataChecker(cellType: CellType) extends NoDataValueChecker[ByteArrayTile] {
     override def doCheck(nodata: Option[Any]): Unit = {
@@ -152,6 +170,9 @@ object AvroTools {
   implicit class UByteNoDataValueCheckMethods(val self: UByteArrayTile) extends
     UByteNoDataChecker(self.cellType) with AvroNoDataCheckMethods[UByteArrayTile] {}
 
+  implicit class UByteConstantNoDataValueCheckMethods(val self: UByteConstantTile) extends
+    UByteNoDataChecker(self.cellType) with AvroNoDataCheckMethods[UByteConstantTile] {}
+
   class UByteNoDataChecker(cellType: CellType) extends NoDataValueChecker[UByteArrayTile] {
     override def doCheck(nodata: Option[Any]): Unit = {
       cellType match {
@@ -165,6 +186,9 @@ object AvroTools {
 
   implicit class BitNoDataValueCheckMethods(val self: BitArrayTile) extends
     BitNoDataChecker(self.cellType) with AvroNoDataCheckMethods[BitArrayTile] {}
+
+  implicit class BitConstantNoDataValueCheckMethods(val self: BitConstantTile) extends
+    BitNoDataChecker(self.cellType) with AvroNoDataCheckMethods[BitConstantTile] {}
 
   class BitNoDataChecker(cellType: CellType) extends NoDataValueChecker[BitArrayTile] {
     override def doCheck(nodata: Option[Any]): Unit = {

--- a/spark/src/test/scala/geotrellis/spark/io/avro/ConstantTileCodecsSpec.scala
+++ b/spark/src/test/scala/geotrellis/spark/io/avro/ConstantTileCodecsSpec.scala
@@ -1,0 +1,119 @@
+/*
+ * Copyright 2016 Azavea
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package geotrellis.spark.io.avro
+
+import geotrellis.spark.io.avro.codecs.TileCodecs
+import org.scalatest._
+import TileCodecs._
+import geotrellis.raster._
+import geotrellis.spark.io.avro.AvroTools._
+
+class ConstantTileCodecsSpec extends FunSpec with Matchers with AvroTools  {
+  describe("ConstantTileCodecs") {
+    it("encodes BitConstantTile"){
+      roundTripWithNoDataCheck(BitConstantTile(1,10,15))
+    }
+    it("encodes ByteConstantTile"){
+      roundTripWithNoDataCheck(ByteConstantTile(127,10,15))
+    }
+    it("encodes UByteConstantTile"){
+      roundTripWithNoDataCheck(UByteConstantTile(127,10,15))
+    }
+    it("encodes ShortConstantTile"){
+      roundTripWithNoDataCheck(ShortConstantTile(45,10,15))
+    }
+    it("encodes UShortConstantTile"){
+      roundTripWithNoDataCheck(UShortConstantTile(45,10,15))
+    }
+    it("encodes IntConstantTile"){
+      roundTripWithNoDataCheck(IntConstantTile(45,10,15))
+    }
+    it("encodes FloatConstantTile"){
+      roundTripWithNoDataCheck(FloatConstantTile(532.4f,10,15))
+    }
+    it("encodes DoubleConstantTile"){
+      roundTripWithNoDataCheck(DoubleConstantTile(53232322.4,10,15))
+    }
+
+    it("encodes ArrayMultibandTile"){
+      val tiles= for (i <- 0 to 3) yield DoubleConstantTile(53232322.4,10,15)
+      val thing = ArrayMultibandTile(tiles): MultibandTile
+      roundTripWithNoDataCheck(thing)
+//      val bytes = AvroEncoder.toBinary(thing)
+//      val fromBytes = AvroEncoder.fromBinary[MultibandTile](bytes)
+//      fromBytes should be equals (thing)
+    }
+  }
+
+  describe("No NoData TileCodecs") {
+    it("encodes ByteConstantTile"){
+      roundTripWithNoDataCheck(ByteConstantTile(127,10,15, ByteCellType))
+    }
+    it("encodes UByteConstantTile"){
+      roundTripWithNoDataCheck(UByteConstantTile(127,10,15, UByteCellType))
+    }
+    it("encodes ShortConstantTile"){
+      roundTripWithNoDataCheck(ShortConstantTile(45,10,15, ShortCellType))
+    }
+    it("encodes UShortConstantTile"){
+      roundTripWithNoDataCheck(UShortConstantTile(45,10,15, UShortCellType))
+    }
+    it("encodes IntConstantTile"){
+      roundTripWithNoDataCheck(IntConstantTile(45,10,15, IntCellType))
+    }
+    it("encodes FloatConstantTile"){
+      roundTripWithNoDataCheck(FloatConstantTile(532.4f,10,15, FloatCellType))
+    }
+    it("encodes DoubleConstantTile"){
+      roundTripWithNoDataCheck(DoubleConstantTile(53232322.4,10,15, DoubleCellType))
+    }
+    it("encodes ArrayMultibandTile"){
+      val tiles= for (i <- 0 to 3) yield DoubleConstantTile(53232322.4,10,15, DoubleCellType)
+      val thing = ArrayMultibandTile(tiles): MultibandTile
+      roundTripWithNoDataCheck(thing)
+    }
+  }
+
+  describe("UserDefined TileCodecs") {
+    it("encodes ByteConstantTile"){
+      roundTripWithNoDataCheck(ByteConstantTile(127,10,15, ByteUserDefinedNoDataCellType(123)))
+    }
+    it("encodes UByteConstantTile"){
+      roundTripWithNoDataCheck(UByteConstantTile(127,10,15, UByteUserDefinedNoDataCellType(123)))
+    }
+    it("encodes ShortConstantTile"){
+      roundTripWithNoDataCheck(ShortConstantTile(45,10,15, ShortUserDefinedNoDataCellType(123)))
+    }
+    it("encodes UShortConstantTile"){
+      roundTripWithNoDataCheck(UShortConstantTile(45,10,15, UShortUserDefinedNoDataCellType(123)))
+    }
+    it("encodes IntConstantTile"){
+      roundTripWithNoDataCheck(IntConstantTile(45,10,15, IntUserDefinedNoDataCellType(123)))
+    }
+    it("encodes FloatConstantTile"){
+      roundTripWithNoDataCheck(FloatConstantTile(532.4f,10,15, FloatUserDefinedNoDataCellType(2.2F)))
+    }
+    it("encodes DoubleConstantTile"){
+      roundTripWithNoDataCheck(DoubleConstantTile(53232322.4,10,15, DoubleUserDefinedNoDataCellType(2.2)))
+    }
+    it("encodes ArrayMultibandTile"){
+      val tiles= for (i <- 0 to 3) yield DoubleConstantTile(53232322.4,10,15, DoubleUserDefinedNoDataCellType(42.23))
+      val thing = ArrayMultibandTile(tiles): MultibandTile
+      roundTripWithNoDataCheck(thing)
+    }
+  }
+}

